controller.py

```diff
@@ -86,12 +86,22 @@ def get_docbins(
 
 
 def manage_encoding_thread(project_id: str, embedding_id: str) -> int:
-    daemon.run(prepare_run_encoding, project_id, embedding_id)
+    daemon.run(prepare_run_encoding_thread, project_id, embedding_id)
     return status.HTTP_200_OK
 
 
-def prepare_run_encoding(project_id: str, embedding_id: str) -> None:
+def prepare_run_encoding_thread(project_id: str, embedding_id: str) -> None:
     session_token = general.get_ctx_token()
+    t = None
+    try:
+        t = __prepare_run_encoding(project_id, embedding_id)
+    finally:
+        general.remove_and_refresh_session(session_token)
+    if t:
+        run_encoding(*t)
+
+
+def __prepare_run_encoding(project_id: str, embedding_id: str) -> None:
```

Comment:
As threading is delegated to manage_encoding_thread, perhaps a renaming suggestion? As prepare_run_encoding_thread does not contain any reference to the daemon or thread:

prepare_run_encoding_thread -> prepare_run
__prepare_run_encoding -> __prepare_encoding

Reply:
changed